fastly/fastly_alerts.go

```diff
@@ -0,0 +1,344 @@
+package fastly
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+// AlertDefinition holds the configuration for one alert.
+type AlertDefinition struct {
+	ID                 string              `json:"id"`
+	ServiceID          string              `json:"service_id"`
+	ServiceCustomerID  string              `json:"service_customer_id,omitempty"`
+	Name               string              `json:"name"`
+	Description        string              `json:"description"`
+	Metric             string              `json:"metric"`
+	Source             string              `json:"source"`
+	Dimensions         map[string][]string `json:"dimensions"`
+	EvaluationStrategy map[string]any      `json:"evaluation_strategy"`
+	IntegrationIDs     []string            `json:"integration_ids"`
+	CreatedAt          time.Time           `json:"created_at"`
+	UpdatedAt          time.Time           `json:"updated_at"`
+	CreatedBy          string              `json:"created_by,omitempty"`
+}
+
+// AlertHistory describes the status of an alert definition over a time range.
+type AlertHistory struct {
+	ID           string          `json:"id"`
+	DefinitionID string          `json:"definition_id"`
+	Definition   AlertDefinition `json:"definition"`
+	Start        time.Time       `json:"start"`
+	End          time.Time       `json:"end"`
+	Status       string          `json:"status"`
+}
+
+// AlertDefinitionsResponse is the response for an alert definitions query.
+type AlertDefinitionsResponse struct {
+	Data []AlertDefinition `json:"data"`
+	Meta AlertsMeta        `json:"meta"`
+}
+
+// AlertHistoryResponse is the response for an alert history query.
+type AlertHistoryResponse struct {
+	Data []AlertHistory `json:"data"`
+	Meta AlertsMeta     `json:"meta"`
+}
+
+// AlertsMeta holds metadata about an alerts query.
+type AlertsMeta struct {
+	NextCursor string `json:"next_cursor"`
+	Limit      int    `json:"limit"`
+	Sort       string `json:"sort"`
+	Total      int    `json:"total"`
+}
+
+// ListAlertDefinitionsInput is used as input to the ListAlertDefinitions function.
+type ListAlertDefinitionsInput struct {
+	// ServiceID filters definitions by service (optional).
+	ServiceID *string
+	// ServiceCustomerID is for Fastly Staff use only (optional).
+	ServiceCustomerID *string
+	// Name filters definitions by name substring (optional).
+	Name *string
+	// CreatedBy is for Fastly Staff use only (optional).
+	CreatedBy *string
+	// Cursor is the pagination cursor from a previous request's meta (optional).
+	Cursor *string
+	// Limit is the maximum number of items included in each response (optional).
+	Limit *int
+	// Sort is the field on which to sort definitions (optional).
+	Sort *string
```

Comment:
Please ensure all fields are sorted alphabetically. This makes it consistent with the rest of the codebase and also easier to search for fields (as a user). Please apply this request to **_all_** structs. Thanks!

Reply:
Fixed, for ***all*** structs.